### PR TITLE
client code clean up

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

File: src/cpp/client/channel.cc [hunk 1]
```diff
@@ -61,19 +61,25 @@ Channel::~Channel() { grpc_channel_destroy(c_channel_); }
...
-                          context->propagation_options_.c_bitmask(), cq->cq(),
-                          method.channel_tag(), context->raw_deadline())
-                    : grpc_channel_create_call(
-                          c_channel_, context->propagate_from_call_,
-                          context->propagation_options_.c_bitmask(), cq->cq(),
-                          method.name(), context->authority().empty()
-                                             ? host_str
-                                             : context->authority().c_str(),
-                          context->raw_deadline());
+  bool registered = method.channel_tag() && context->authority().empty();
```
Comment (dgquintas) [inline src/cpp/client/channel.cc]: const

File: src/cpp/client/client_context.cc [hunk 2]
```diff
@@ -48,7 +48,6 @@ namespace grpc {
 ClientContext::ClientContext()
     : initial_metadata_received_(false),
       call_(nullptr),
-      cq_(nullptr),
```
Comment (dgquintas) [inline src/cpp/client/client_context.cc]: out of curiosity, why don't we need/use cq_ anymore?

**Comment by dgquintas** [comment]
LGTM other than the "const" nit.

File: src/cpp/client/client_context.cc (hunk 2 above)
Comment (yang-g) [inline src/cpp/client/client_context.cc]: I think it was lost in refactors...

File: src/cpp/client/channel.cc (hunk 1 above)
Comment (yang-g) [inline src/cpp/client/channel.cc]: Done.

**Pull request merged**